src/Xamarin.Android.Build.Tasks/Linker/MonoDroid.Tuner/MonoDroidMarkStep.cs

```diff
@@ -128,31 +143,112 @@ MethodReference CreateGenericMethodReference (MethodReference method, GenericIns
 			return genericMethod;
 		}
 
-		void UpdateRegistrationSwitch (MethodDefinition method, MethodReference[] switchMethods)
+		// should stay in sync with AndroidRuntime.AndroidTypeManager.MagicRegistrationMap.GetStringHashCode
+		static int GetStringHashCode (string str)
+                {
+                        int hash1 = 5381;
```

Comment:
Anyway we can centralise this so we don't need to update it in two places?

Reply:
I was thinking about that and we can probably put it somewhere in Java.Interop as it is referenced by both projects.

Reply:
It is still proof of concept and needs improvements